opts.go

```diff
@@ -47,3 +49,20 @@ func WithProcessRunner(cmd *exec.Cmd) Opt {
 		machine.cmd = cmd
 	}
 }
+
+// WithSnapshot will allow for the machine to start using a given snapshot.
+func WithSnapshot(ctx context.Context, memFilePath, snapshotPath string, opts ...LoadSnapshotOpt) Opt {
+	return func(m *Machine) {
+		snapshotParams := &models.SnapshotLoadParams{
+			MemFilePath:  String(memFilePath),
+			SnapshotPath: String(snapshotPath),
+		}
+
+		if _, err := m.client.LoadSnapshot(ctx, snapshotParams, opts...); err != nil {
+			m.logger.Errorf("failed to load a snapshot for VM: %v", err)
+			return
+		}
+
```

Comment:
If err != nil, I think Start() should return the error.

Right now, the tests are passing but logging

```
time="2022-06-22T18:18:23Z" level=error msg="failed to load a snapshot for VM: Put \"http://localhost/snapshot/load\": dial unix /tmp/TestCreateLoadSnapshot688360910/TestCreateLoadSnapshot.load: connect: no such file or directory"
```

Reply:
I understand and think we should find a way to do that, but the current `Opt` type doesn't return anything, so we would have to change the type to return an error. It doesn't seem that the scope of this change would be a big deal (currently only seeing references to the type within `opts.go`), but as these functions are exposed, in theory it could break something as we're moving from returning nothing to returning an error. I think this is alright, but wanted to make sure you agreed before moving forward with the changes.

I feel this change would allow for better error handling down the line should we add any more `WithFoo`-esque commands that require more complex computations than setting a variable; however, the above concern still remains.

An alternative approach is to add some machine variables, have `WithSnapshot` set aforementioned parameters, then checks if those parameters have been set after all the options are called, then call `client.LoadSnapshot` with said parameters if so.

Reply:
Well does it make sense to have StartOpt instead of reusing Opt? NewMachine wouldn't take WithSnapshot and Start wouldn't take WithProcessRunner.

Reply:
That makes sense, yes. I'll make the adjustments and do some debugging once I figure out why the snapshot is failing to load.

Reply:
Deleted my previous comment as it contained incorrect information.

@austinvazquez and I combed through the code and figured out that snapshots were not loading with the previous implementation due to the lack of a vsock when attempting to load the snapshot. However, trying to do so after the handler is not allowed, so we had to make the snapshot loader a handler that would be dynamically inserted before the handler that starts the VMM.

I also dropped the requirement for `StartOpt()` to return an error, as it now only appends to the handler array. If `LoadSnapshot()` itself fails, it will be handled via `m.Handlers.Run()` now.

It seems to pass the tests after these changes, as the handler's `Run()` function requires no errors, but when adding metadata to a machine before creating the snapshot, it is not loaded onto the snapshot, leading me to believe that either snapshots do not preserve metadata or `LoadSnapshot()` is somehow not functioning correctly on the Firecracker side, so I presume in the scope of this problem this is not something I should worry about, but feel free to correct me if my assumption is wrong.